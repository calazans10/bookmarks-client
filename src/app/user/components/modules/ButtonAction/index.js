import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './index.scss';

const ButtonAction = ({ kind, onClick, children }) => {
  const btnClass = classNames({
    'button-action': true,
    [`button-action--${kind}`]: true,
  });

  return (
    <button type="button" className={btnClass} onClick={onClick}>
      {children}
    </button>
  );
};

ButtonAction.propTypes = {
  kind: PropTypes.oneOf(['success', 'danger']).isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default ButtonAction;
