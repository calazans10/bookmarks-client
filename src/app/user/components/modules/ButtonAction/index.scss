@import 'styles/utilities';

.button-action {
  width: rem(100);
  padding: rem(8);
  border: none;
  color: #fff;
  font-size: em(10);
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  cursor: pointer;

  &:focus,
  &:hover {
    color: #fff;
  }

  &--success {
    background-color: #27ae60;
  }

  &--danger {
    margin-left: rem(5);
    background-color: #e74c3c;
  }
}
