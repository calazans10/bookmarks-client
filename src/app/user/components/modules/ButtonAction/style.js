import styled from 'styled-components';
import { em, rem } from 'polished';

const theme = {
  success: '#27ae60',
  danger: '#e74c3c',
};

export const Button = styled.button`
  width: ${rem('100px')};
  padding: ${rem('8px')};
  background-color: ${props => theme[props.theme.main]};
  border: none;
  color: #fff;
  font-size: ${em('10px')};
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  cursor: pointer;

  &:focus,
  &:hover {
    color: #fff;
  }

  &:last-of-type {
    margin-left: ${rem('5px')};
  }
`;
