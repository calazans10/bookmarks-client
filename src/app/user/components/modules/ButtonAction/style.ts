import styled from 'styled-components';
import { em, rem } from 'polished';

const theme = {
  success: '#27ae60',
  danger: '#e74c3c',
};

type ButtonTheme = {
  main: 'success' | 'danger';
};

type ButtonProps = {
  theme: ButtonTheme;
};

export const Button = styled.button<ButtonProps>`
  width: ${rem('100px')};
  padding: ${rem('8px')};
  border: none;
  background-color: ${(props: ButtonProps) => theme[props.theme.main]};
  color: #fff;
  font-size: ${em('10px')};
  font-weight: 500;
  letter-spacing: 2px;
  text-transform: uppercase;
  cursor: pointer;

  &:focus,
  &:hover {
    color: #fff;
  }

  &:last-of-type {
    margin-left: ${rem('5px')};
  }
`;
