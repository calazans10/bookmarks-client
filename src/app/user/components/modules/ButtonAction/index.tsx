import React from 'react';
import PropTypes from 'prop-types';
import { Button } from './style';

const ButtonAction = ({ kind, onClick, children }) => {
  return (
    <Button theme={{ main: kind }} type="button" onClick={onClick}>
      {children}
    </Button>
  );
};

ButtonAction.propTypes = {
  kind: PropTypes.oneOf(['success', 'danger']).isRequired,
  onClick: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};

export default ButtonAction;
