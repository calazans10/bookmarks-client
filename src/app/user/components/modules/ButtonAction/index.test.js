import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ButtonAction from './index';

describe('ButtonAction', () => {
  it('renders without crashing', () => {
    const props = {
      kind: 'success',
      onClick: jest.fn(),
    };
    const wrapper = shallow(<ButtonAction {...props}>Edit</ButtonAction>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
