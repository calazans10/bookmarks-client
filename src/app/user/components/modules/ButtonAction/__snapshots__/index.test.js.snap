// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonAction renders without crashing 1`] = `
<button
  class="sc-AykKC kTcdEt"
  type="button"
>
  Edit
</button>
`;
